### ISSUE TITLE
Key and nonce should have separate lengths

### ISSUE THREAD

**Issue by bifurcation** [mlswg/mls-protocol#117]
This text is wrong:

```
   key = HKDF-Expand(Secret, ECIESLabel("key"), Length)
   nonce = HKDF-Expand(Secret, ECIESLabel("nonce"), Length
```

**Comment by bifurcation** [comment]
This could also be addressed by using draft-barnes-cfrg-hpke.

**Comment by beurdouche** [comment]
This has been fixed by #116
